mono/utils/mono-threads-windows.c

```diff
@@ -29,13 +29,12 @@ mono_threads_suspend_begin_async_suspend (MonoThreadInfo *info, gboolean interru
 	HANDLE handle;
 	DWORD result;
 
-	handle = OpenThread (THREAD_ALL_ACCESS, FALSE, id);
+	handle = info->native_handle;
```

Comment:
We need to use open thread to ensure handle safety.

Reply:
Please ask to restore any missing APIs you need. APIs are gradually being added back as needed.
OpenThread is already there:
https://msdn.microsoft.com/en-us/library/windows/desktop/ms684335(v=vs.85).aspx

Reply:
It was only added in the latest SDK. We still support older ones.

Reply:
That is unfortunate. Is there another way? Can you autoconf-like it? Or bind dynamically and always find it?

Reply:
Why is it unfortunate? Isn't the non-OpenThread way superior anyway because it just reads a value rather than calling a Windows API?

Reply:
The issue is around safety. I think supporting old sdks is a good thing for us to have.

Reply:
The hazard here is that the handle might become asynchronously invalid.
Using OpenThread ensures we increment the handle ref count before using it.
This is not an issue on unix because we have to pthread_join to free the handle.

I think your solution should of dup'ing on mono_threads_suspend_register should work.
I'll let @lateralusX weight in, but I'm good with how you solved it.

Reply:
Agree, doing the duping on mono_threads_suspend_register and then close in mono_threads_suspend_free should keep the thread alive during the above method calls. The above methods will be called from register_thread and unregister_thread connected to the overall lifetime of the thread from mono runtimes perspective.

Thread lifetime management has on the other hand been a problematic area in the past, especially during shutdown scenarios, but has been improved over the last year, so we need to pay attention to sporadic test case failures popping up after changes like this. For this PR tests runs looks ok 100% pass on x86 and x64 so that is a good start.

Reply:
So far I think we're still good, without any test case failures. Can this go in as-is then?